docs/book/src/migration/plugin/v2_v3.md

```diff
@@ -1,3 +1,570 @@
+
+The projects built with v3+ plugin now define a specific `user ID` and `securityContext` policy to prevent root privilege escalation from the manager container.
+
+- Update your Dockerfile to define an ID for the user used:
+
+Replace:
+
+```
+USER nonroot:nonroot
+```
+
+With:
+
+```
+USER 65532:65532
+```
+
+- Update your `manager.yaml` manifests in the `config/manager/` directory by adding:
+
+```yaml
+...
+spec:
+      securityContext:
+        runAsUser: 65532
+...
+```
+
+```yaml
+...
+name: manager
+securityContext:
+  allowPrivilegeEscalation: false
+
+```
+
+The final result would look like:
+
+```yaml
+...
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: controller-manager
+  namespace: system
+  labels:
+    control-plane: controller-manager
+spec:
+  selector:
+    matchLabels:
+      control-plane: controller-manager
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        control-plane: controller-manager
+    spec:
+      securityContext:
+        runAsUser: 65532
+      containers:
+      - command:
+        - /manager
+        args:
+        - --enable-leader-election
+        image: controller:latest
+        name: manager
+        securityContext:
+          allowPrivilegeEscalation: false
+        resources:
+          limits:
+            cpu: 100m
+            memory: 30Mi
+          requests:
+            cpu: 100m
+            memory: 20Mi
+      terminationGracePeriodSeconds: 10
+```
+
+### Add the Liveness and Readiness probes
+
+From now on, the new projects build with the tool will be scaffolding Liveness and Readiness probes using [`healthz.Ping`][healthz-ping]. To add them to your project update your `main.go file:
+
+- By adding the new flag `--health-probe-bind-address`:
+
+```go
+func main() {
+	var metricsAddr string
+	var enableLeaderElection bool
+	var probeAddr string
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	...
+``` 
+
+- By using the new flag in the manager: 
+
+```go 
+mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:                 scheme,
+		MetricsBindAddress:     metricsAddr,
+		Port:                   9443,
+		HealthProbeBindAddress: probeAddr,
+		LeaderElection:         enableLeaderElection,
+		LeaderElectionID:       "dd1da13f.testproject.org",
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to start manager")
+		os.Exit(1)
+	}
+```
+
+- Then, add the new probe endpoints after the Manager is initialized:
+
+```go
+
+	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+		log.Error(err, "Unable to set up health check")
+		os.Exit(1)
+	}
+
+	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
+		log.Error(err, "Unable to set up ready check")
+		os.Exit(1)
+	}
+
+```
+
+Finally, update the `config/default/manager.yaml` to add the probes:
```

Comment:
Should this be `config/manager/manager.yaml` instead?

Reply:
/test if u think it's good then go for it and test it 
M

Reply:
I need to fix that. You are right it is `config/manager/manager.yaml`. I will do that.

Reply:
solved